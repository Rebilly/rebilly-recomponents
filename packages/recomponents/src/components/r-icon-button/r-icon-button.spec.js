import {renderToString} from '@vue/server-test-utils';
import {shallowMount} from '@vue/test-utils';
import RIconButton from './r-icon-button.vue';

describe('r-icon-button.vue', () => {
    it('should render via SSR and match snapshot', async () => {
        const wrapper = await renderToString(RIconButton, {});

        expect(wrapper).toMatchSnapshot();
    });

    it('should render component with capitalizing the first letter', async () => {
        const wrapper = shallowMount(RIconButton, {
            slots: {
                default: 'click me',
            },
        });

        expect(wrapper.text()).toMatch('Click me');
    });
});
