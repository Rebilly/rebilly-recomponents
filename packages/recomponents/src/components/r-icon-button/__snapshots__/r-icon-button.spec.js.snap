// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`r-icon-button.vue should render via SSR and match snapshot 1`] = `<button class="r-button r-button--type-default r-button--size-regular has-icon"> </button>`;
